crates/runtime/src/lib.rs

```diff
@@ -93,49 +95,81 @@ impl Runtime {
     #[must_use]
     pub fn new(
         config: Config,
-        app: app::App,
-        df: DataFusion,
+        app: Arc<app::App>,
+        df: Arc<RwLock<DataFusion>>,
         models: HashMap<String, Model>,
         pods_watcher: podswatcher::PodsWatcher,
     ) -> Self {
         Runtime {
-            app: Arc::new(app),
+            app,
             config,
-            df: Arc::new(RwLock::new(df)),
+            df,
             models: Arc::new(models),
             pods_watcher,
         }
     }
 
-    pub async fn load_dataset(
-        ds: &Dataset,
-        df: &mut datafusion::DataFusion,
-        auth: &auth::AuthProviders,
-    ) -> Result<()> {
-        let ds = Arc::new(ds.clone());
-        if ds.acceleration.is_none() && !ds.is_view() {
-            tracing::warn!("No acceleration specified for dataset: {}", ds.name);
-            return Ok(());
-        };
-
-        let source = ds.source();
-        let source = source.as_str();
-        let params = Arc::new(ds.params.clone());
-        let data_connector: Option<Box<dyn DataConnector>> =
-            Runtime::get_dataconnector_from_source(source, auth, Arc::clone(&params)).await?;
-
-        Runtime::initialize_dataconnector(data_connector, df, source, &ds).await?;
+    pub fn load_datasets(&self, auth: &Arc<auth::AuthProviders>) {
+        for ds in self.app.datasets.clone() {
+            self.load_dataset(&ds, auth);
+        }
+    }
 
-        tracing::info!("Loaded dataset: {}", &ds.name);
+    pub fn load_dataset(&self, ds: &Dataset, auth: &Arc<auth::AuthProviders>) {
+        let ds = ds.clone();
+        let df = self.df.clone();
+        let auth = auth.clone();
```

Comment:
One reason I prefer using Arc::clone vs .clone() is that for example here - which ones are just cloning a pointer vs cloning the actual object?

Reply:
Ugh I know but it feels so much nice to just call .clone!

Will update throughout the PR :)